gensrc/thrift/DataSinks.thrift

```diff
@@ -213,6 +214,7 @@ struct TIcebergTableSink {
     4: optional Types.TTupleId tuple_id
     5: optional Types.TCompressionType compression_type
     6: optional bool is_statistics_partition_sink
```

Comment:
Newly added fields should be optional, please change the first member `location` to optional

Reply:
`TIcebergTableSink` as a whole is an optional field in `TDataSink`.

Also, FE has to give BE a location to specify where to write data files. I think `required` is semantically correct here.

Reply:
In starrocks, all fields in thrift/protobuf need to be optional. Currently, it may be *required*, but it may become optional someday. You can see there are so many Required fields that are deprecated now.
So please make them optional. This is one rule.

Reply:
fixed